test-framework/clustering/src/main/java/org/keycloak/testframework/clustering/LoadBalancerSupplier.java

```diff
@@ -0,0 +1,27 @@
+package org.keycloak.testframework.clustering;
+
+import org.keycloak.testframework.annotations.InjectLoadBalancer;
+import org.keycloak.testframework.injection.InstanceContext;
+import org.keycloak.testframework.injection.RequestedInstance;
+import org.keycloak.testframework.injection.Supplier;
+import org.keycloak.testframework.server.ClusteredKeycloakServer;
+import org.keycloak.testframework.server.KeycloakServer;
+
+public class LoadBalancerSupplier implements Supplier<LoadBalancer, InjectLoadBalancer> {
+
+    @Override
+    public LoadBalancer getValue(InstanceContext<LoadBalancer, InjectLoadBalancer> instanceContext) {
+        KeycloakServer server = instanceContext.getDependency(KeycloakServer.class);
+
+        if (!(server instanceof ClusteredKeycloakServer)) {
+            throw new IllegalStateException("Load balancer can only be used with ClusteredKeycloakServer");
+        }
+
+        return new LoadBalancer((ClusteredKeycloakServer) server);
```

Comment:
Nitpick, but I like this approach:
```suggestion
        if (server instanceof ClusteredKeycloakServer clusteredServer) {
            return new LoadBalancer(clusteredServer);
        } else {
             throw new IllegalStateException("Load balancer can only be used with ClusteredKeycloakServer");        
        }
```